packaging/installer/netdata-updater.sh

```diff
@@ -73,6 +73,152 @@ if [ "${ENVIRONMENT_FILE}" = "THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT" ]; th
   fi
 fi
 
+issystemd() {
+  # if the directory /lib/systemd/system OR /usr/lib/systemd/system (SLES 12.x) does not exit, it is not systemd
+  if [ ! -d /lib/systemd/system ] && [ ! -d /usr/lib/systemd/system ]; then
+    return 1
+  fi
+
+  # if there is no systemctl command, it is not systemd
+  systemctl=$(command -v systemctl 2> /dev/null)
+  if [ -z "${systemctl}" ] || [ ! -x "${systemctl}" ]; then
+    return 1
+  fi
+
+  # if pid 1 is systemd, it is systemd
+  [ "$(basename "$(readlink /proc/1/exe)" 2> /dev/null)" = "systemd" ] && return 0
+
+  # if systemd is not running, it is not systemd
+  pids=$(safe_pidof systemd 2> /dev/null)
+  [ -z "${pids}" ] && return 1
+
+  # check if the running systemd processes are not in our namespace
+  myns="$(readlink /proc/self/ns/pid 2> /dev/null)"
+  for p in ${pids}; do
+    ns="$(readlink "/proc/${p}/ns/pid" 2> /dev/null)"
+
+    # if pid of systemd is in our namespace, it is systemd
+    [ -n "${myns}" ] && [ "${myns}" = "${ns}" ] && return 0
+  done
+
+  # else, it is not systemd
+  return 1
+}
+
+_get_scheduler_type() {
+  if _get_intervaldir > /dev/null ; then
+    echo 'interval'
+  elif issystemd ; then
+    echo 'systemd'
+  elif [ -d /etc/cron.d ] ; then
+    echo 'crontab'
+  else
+    echo 'none'
+  fi
+}
```

Comment:
Why this order? Not a specialist, but I guess "systemd" should be either the first or the last.

Reply:
Because systemd can’t give failure notifications.

Reply:
I see, then why do we prefer systemd over /etc/cron.d?

Reply:
Because it’s more likely to be available, and also less likely to appear to be present but not actually work.

Reply:
Ok, thanks for the explanation. I don't really get it:
 - we prefer cron.daily over systemd, because systemd can't give failure notifications.
 - but we prefer systemd over crontab regardless the same problem with systemd remains.

Not a specialist, was just curious.

Reply:
In an ideal situation, we both want to be sure it works and that it gives failure notifications, but being sure it works is more important than the failure notifications.

The cron.daily approach is about 95% certain to work if we see the infrastructure for it on the system already, and it provides failure notifications.

The systemd approach is roughly 100% certain to work if we find it, but does not give failure notifications.

The crontab approach is roughly 80% certain to work if we find it, and does give failure notifications.

Also though, the cron.daily approach was the only one we supported up until about a year ago, so we kept that as the default to avoid surprising users.